src/cairo1-compiler.ts

```diff
@@ -0,0 +1,100 @@
+import { ProcessResult } from "@nomiclabs/hardhat-docker";
+import shell from "shelljs";
+import { Image } from "@nomiclabs/hardhat-docker";
+import { DockerServer } from "./external-server/docker-server";
+import { getFreePort } from "./external-server/external-server";
+import { CommonSpawnOptions } from "child_process";
+
+export const exec = (args: string) => {
+    const result = shell.exec(args);
+    return {
+        statusCode: result.code,
+        stdout: Buffer.from(result.stderr),
+        stderr: Buffer.from(result.stdout)
+    } as ProcessResult;
+};
+
+export class DockerCairo1Compiler extends DockerServer {
```

Comment:
Why are we spawning a separate container with a server here? Which server will it be running?

Reply:
it'll create a short-lived process to run the compilation of cairo1 contracts on dockerized mode. It derives `DockerServer` class to leverage the `spawnChildProcess` function. Needed a separate container to  run two commands at a time and with a guaranteed free port.

Reply:
Was thinking of the above reasons, but now I think about it more, and that the container will have a unique name for each compilation a server wouldn't be necessary. https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/328/commits/dd9a11fbd8ed11c4882a99ed4e63dbf9e6723b71

Reply:
but it's still inheriting from DockerServer, right?

Reply:
Yes. It still inherits from it because of the required funcs `getDockerArgs` and `getContainerArgs`

Reply:
will be addressed in a separate PR